spec/support/win32.cr

```diff
@@ -17,3 +17,24 @@ require "spec"
     describe(describe, file, line, end_line, &block)
   end
 {% end %}
+
+# Heuristic to determine the current MSYS2 environment:
+#
+# - target triple must be `*-windows-gnu`
+# - `uname -s` must return `MINGW64_NT*` or `MSYS_NT*`
+# - the `MSYSTEM` environment variable must be non-empty
+#
+# (ideally we should determine the C runtime used at build time, for now we just
+# assume the build-time and run-time environments are identical)
+def msys2_environment : String?
+  {% if flag?(:win32) && flag?(:gnu) %}
+    uname = IO::Memory.new
+    if Process.run("uname", %w(-s), output: uname).success?
+      uname = uname.to_s
+      if uname.starts_with?("MINGW64_NT") || uname.starts_with?("MSYS_NT")
+        ENV["MSYSTEM"]?.presence
+      end
+    end
+  {% end %}
+rescue IO::Error
```

Comment:
**question:** When would the code in this method raise `IO::Error`?

Reply:
when `uname` doesn't exist

Reply:
Right. `File::NotFoundError` from `Process.run`. I was focused on `IO::Memory`.